libs/src/sdk/services/Storage/Storage.ts

```diff
@@ -0,0 +1,116 @@
+import FormData from 'form-data'
+import axios from 'axios'
+
+import fetch from 'cross-fetch'
+import { wait } from '../../utils/wait'
+import type {
+  FileTemplate,
+  ProcessingStatus,
+  ProgressCB,
+  StorageService,
+  StorageServiceConfig
+} from './types'
+import { mergeConfigWithDefaults } from '../../utils/mergeConfigs'
+import {
+  defaultStorageServiceConfig,
+  MAX_IMAGE_RESIZE_TIMEOUT_MS,
+  MAX_TRACK_TRANSCODE_TIMEOUT,
+  POLL_STATUS_INTERVAL
+} from './constants'
+import type { CrossPlatformFile as File } from '../../types/File'
+import { isNodeFile } from '../../utils/file'
+
+export class Storage implements StorageService {
+  /**
+   * Configuration passed in by consumer (with defaults)
+   */
+  private config: StorageServiceConfig
+
+  constructor(config?: StorageServiceConfig) {
+    this.config = mergeConfigWithDefaults(config, defaultStorageServiceConfig)
+  }
+
+  /**
+   * Upload a file to a content node
+   * @param file
+   * @param onProgress
+   * @param template
+   * @returns
+   */
+  async uploadFile({
+    file,
+    onProgress,
+    template
+  }: {
+    file: File
+    onProgress: ProgressCB
+    template: FileTemplate
+  }) {
+    const formData: FormData = new FormData()
+    formData.append('template', template)
+    // TODO: Test this in a browser env
+    formData.append('files', isNodeFile(file) ? file.buffer : file, file.name)
+
+    // Using axios for now because it supports upload progress,
+    // and Node doesn't support XmlHttpRequest
+    const response = await axios({
```

Comment:
@rickyrombo I went down the XHR route but node doesn't support it so there still ended up being fragmentation. Decided to stick with axios for this request for now, since we need upload progress

Reply:
OOOOOOF